.github/workflows/main.yml

```diff
@@ -41,21 +28,32 @@ jobs:
         run: |
           make nodeup examples test
 
-  verify:
-    strategy:
-      matrix:
-        os: [ubuntu-18.04]
-        go: [1.13]
-      fail-fast: true
+  build-macos-amd64:
+    runs-on: macos-10.15
+    steps:
+    - name: Set up go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14
+        stable: true
+
+    - uses: actions/checkout@v2
+      with:
+        path: ${{ env.GOPATH }}/src/k8s.io/kops
 
-    runs-on: ${{ matrix.os }}
+    - name: make nodeup examples test
+      working-directory: ${{ env.GOPATH }}/src/k8s.io/kops
+      run: |
+        make nodeup examples test
```

Comment:
We can probably skip making nodeup on MacOS.

Reply:
Good point. Done.

Reply:
Any idea why we only build `nodeup` and not `all` ?

Reply:
I suspect this is just historical about the way these tests were introduced (they're still fairly new); I suspect we were just trying to replace tests in travis.

From that point of view I really like splitting out the tests explicitly.  What we really want on macos is coverage of the kops CLI (I don't think we want nodeup tests to run on macos; there's a case for windows in the future but I don't know of anyone trying to get kubelet to run on macos)

For linux, I think trying everything seems reasonable!  We should probably be trending towards running tests in bazel, as more and more of our release pipeline is using bazel (I have some exciting news on this front for our next office hours :-) )

I think it's fine to rearrange what these do a bit in a separate PR.

Reply:
Looking forward to that :).